src/linear_eq.jl

```diff
@@ -0,0 +1,217 @@
+"""
+Preconditions the matrix A and b with the inverse of mid(A)
+"""
+function preconditioner(A::AbstractMatrix, b::AbstractArray)
+
+    Aᶜ = mid.(A)
+    B = inv(Aᶜ)
+
+    return B*A, B*b
+
+end
+
+function gauss_seidel_interval(A::AbstractMatrix, b::AbstractArray; precondition=true, maxiter=100)
+
+    n = size(A, 1)
+    x = fill(-1e16..1e16, n)
+
+    if (typeof(b) <: SArray || typeof(b) <: MArray)
+        x = MVector{n}(x)
+    end
+
+    gauss_seidel_interval!(x, A, b, precondition=precondition, maxiter=maxiter)
+    return x
+end
+"""
+Iteratively solves the system of interval linear
+equations and returns the solution set. Uses the
+Gauss-Seidel method (Hansen-Sengupta version) to solve the system.
+Keyword `precondition` to turn preconditioning off.
+Eldon Hansen and G. William Walster : Global Optimization Using Interval Analysis - Chapter 5 - Page 115
+"""
+function gauss_seidel_interval!(x::AbstractArray, A::AbstractMatrix, b::AbstractArray; precondition=true, maxiter=100)
+
+    precondition && ((A, b) = preconditioner(A, b))
+
+    n = size(A, 1)
+
+    @inbounds for iter in 1:maxiter
+        for i in 1:n
+            Y = b[i]
+            for j in 1:n
+                (i == j) || (Y -= A[i, j] * x[j])
+            end
+            Z = extended_div(Y, A[i, i])
+            x[i] = hull((x[i] ∩ Z[1]), x[i] ∩ Z[2])
+        end
+    end
+    x
+end
+#
```

Comment:
All of this should be deleted instead of commented out.
It can be recovered from the git history if necessary.